slot.go

```diff
@@ -75,6 +77,43 @@ func (s *Slot) Key(id string) (*Key, error) {
 	})
 }
 
+// GetIDByLabel returns a hex-encoded ID for the given label
+func (s *Slot) GetIDByLabel(label string) (string, error) {
```

Comment:
I couldn’t just search for the key material matched by a label because the private key material was on the same ID as the public key material, but it didn’t turn up from a search for label. I think I could easily make this into a `KeyFromLabel` function that returned an error if not all of the key material exists.